src/celeritas/UnitTypes.hh

```diff
@@ -0,0 +1,291 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/UnitTypes.hh
+//! \brief Annotated unit struct definitions for use with Quantity
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "celeritas_config.h"
+#include "corecel/math/UnitUtils.hh"
+
+#include "Constants.hh"
+#include "Units.hh"
+
+namespace celeritas
+{
+namespace units
+{
+//---------------------------------------------------------------------------//
+//!@{
+//! \name Natural units
+
+struct CLight
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return constants::c_light;
+    }
+    static char const* label() { return "c"; }
+};
+
+//! "Natural units" for energy
+struct Mev
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+#if CELERITAS_UNITS == CELERITAS_UNITS_CLHEP
```

Comment:
(Opening a new conversation to ask more clearly a question that was implicit in https://github.com/celeritas-project/celeritas/pull/1076/files#r1439655776:

Should the configuration depend code (i.e. `if constexpr` or `#if`) but only in `Units.hh` or also in `UnitsTypes.hh` (it feels that we could make it to be 'jiust' `Units.hh`).

Reply:
It can't be *just* `Units.hh` because the non-CLHEP units require universal constants (electron charge) to compute this value.